### PR TITLE
Test Kitchen Config Refactor

### PR DESCRIPTION
**Pull request by zwall-bp**
Errors present in Vagrant and Dokken

### CODE REVIEW

File: kitchen.dokken.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
+driver:
+  name: dokken
+  privileged: true  # because Docker and SystemD/Upstart
+  chef_image: cincproject/cinc
+  chef_version: <%= ENV['CHEF_VERSION'] || '17' %>
```
Comment (ramereth) [inline kitchen.dokken.yml]:
```suggestion
  chef_version: <%= ENV['CHEF_VERSION'] || '18' %>
```

File: kitchen.yml [hunk 2]
```diff
@@ -1,19 +1,26 @@
...
-driver:
-  name: openstack
-  flavor_ref: 'm1.large'
+transport:
+  name: rsync
 
 provisioner:
   name: chef_infra
+  product_name: cinc
+  product_version: '17'
```
Comment (ramereth) [inline kitchen.yml]:
```suggestion
  product_version: '18'
```

**Review by ramereth — CHANGES_REQUESTED** [review]

**Comment by ramereth** [comment]
@zwall-bp please make sure you rebase this

**Comment by zwall-bp** [comment]
Only issues present are in the `*-runfile-*` resources, in verification, as there are issues parsing for the kernel version.

Ignored dokken tests, as installing drivers goes too low level for Docker's abilities.

**Comment by ramereth** [comment]
@zwall-bp I recommend you just remove the dokken kitchen completely. There's very little this cookbook would benefit from dokken.

File: kitchen.yml (hunk 2 above)
Comment (ramereth) [reply kitchen.yml]: This still needs to be addressed.

**Comment by ramereth** [comment]
@zwall-bp also please add Ubuntu 20.04 to `kitchen.yml`

File: kitchen.openstack.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+transport:
+  name: rsync
+  ssh_key: <%= ENV['OS_PRIVATE_SSH_KEY'] %>
+
+platforms:
+  - name: almalinux-8
+    driver_plugin: openstack
+    driver_config:
+      image_ref: "almalinux 8"
```
Comment (ramereth) [inline kitchen.openstack.yml]:
```suggestion
      image_ref: "AlmaLinux 8"
```

**Pull request merged**